### ISSUE TITLE
Linux no sound ALSA/PulseAudio

### ISSUE THREAD

**Issue by MeindertKempe** [OpenLoco/OpenLoco#416]
**OS:**  Arch Linux
Kernel 5.6.3-arch1-1
GCC 9.3.0
cmake 3.17.1
make 4.3
lib32-SDL2 2.0.12
lib32-SDL2_mixer 2.0.4
yaml-cpp 0.6.3
**Commit/Build:**  8dc249d3676b494fc2aed51c9c7881bd31564af2

I compiled OpenLoco on Arch Linux and as far as I can tell I have no sound at all, when starting OpenLoco I get the following messsages:

    ALSA lib dlmisc.c:283:(snd_dlobj_cache_get0) Cannot open shared library/usr/lib32/alsa-lib/libasound_module_pcm_pulse.so ((null): libpulse.so.0: cannot open shared object file: No such file or directory)

    Mix_OpenAudio failed: ALSA: Couldn't open audio device: No such device or addressCreateRectRgn(0, 0, 1, 1)

That I suspect relate to this issue.

Here is the output I got from compiling and running OpenLoco.
[OpenLoco.txt](https://github.com/OpenLoco/OpenLoco/files/4483188/OpenLoco.txt)

**Is the issue reproducible in Locomotion (vanilla)?**
When running Locomotion in wine I get similar messages as when I'm running OpenLoco and I do not have any sound either.

    0009:err:module:load_so_dll failed to load .so lib "/usr/bin/../lib32/wine/winepulse.drv.so": libpulse.so.0: cannot open shared object file: No such file or directory
    ALSA lib dlmisc.c:283:(snd_dlobj_cache_get0) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_pcm_pulse.so ((null): libpulse.so.0: cannot open shared object file: No such file or directory)
    ALSA lib dlmisc.c:283:(snd_dlobj_cache_get0) Cannot open shared library /usr/lib32/alsa-lib/libasound_module_pcm_pulse.so ((null): libpulse.so.0: cannot open shared object file: No such file or directory)

I have not encountered any other programs with similar sound issues, which leads me to believe it is an issue with OpenLoco.

**Comment by AaronVanGeffen** [comment]
Hmm, this is strange. I run the game on Arch Linux as well, and with no such issues.

How custom is your PulseAudio / ALSA set-up?

**Comment by MeindertKempe** [comment]
I believe my setup is fairly standard.
This is my [asound.conf](https://github.com/OpenLoco/OpenLoco/files/4483366/asound.txt), which I believe is the standard config for using PulseAudio with ALSA.

And when i ran diff on my pulse config files and the default config files I get the folowing:

    diff '--color=auto' pulse/daemon.conf /etc/pulse/daemon.conf
    90a91,94
    > 
    > resample-method = src-sinc-best-quality
    > default-sample-format = s16le
    > default-sample-rate = 48000
    diff '--color=auto' pulse/default.pa /etc/pulse/default.pa
    34a35,37
    > load-module module-alsa-source device=hw:0,0
    > load-module module-alsa-sink device=plughw:0,0
    > 
    47c50
    < load-module module-udev-detect
    ---
    > load-module module-udev-detect use_ucm=0 tsched=0

Which, as far as I can tell isn't particularly unusual.

**Comment by MeindertKempe** [comment]
Solved it, I found [this stackoverflow post](https://stackoverflow.com/questions/37690058/libpulse-so-0-cannot-open-shared-object-file-no-such-file-or-directory), it seems I was missing lib32-pulse. In hindsight it makes sense that this would be required.
I think it may be a good idea to mention the fact that 32 bit alsa/pulse are needed though. (I already had 32bit alsa installed, probably because of steam.)
